webapp/src/ts/services/transitions/create-user-for-contacts.transition.ts

```diff
@@ -0,0 +1,173 @@
+import { Injectable } from '@angular/core';
+
+import { DbService } from '@mm-services/db.service';
+import { Transition, Doc } from '@mm-services/transitions/transition';
+import { CreateUserForContactsService } from '@mm-services/create-user-for-contacts.service';
+import { ExtractLineageService } from '@mm-services/extract-lineage.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CreateUserForContactsTransition extends Transition {
+  constructor(
+    private dbService: DbService,
+    private createUserForContactsService: CreateUserForContactsService,
+    private extractLineageService:ExtractLineageService,
+  ) {
+    super();
+  }
+
+  readonly name = 'create_user_for_contacts';
+  private readonly REPLACE_PROPERTY = 'replace_forms';
+  private replaceForms;
+
+  /**
+   * @param {Object} settings - the CHT instance settings
+   * @return {Boolean} - whether the required config is present
+   */
+  init(settings) {
+    this.replaceForms = this.getReplaceForms(settings);
+    if (!this.replaceForms || !Array.isArray(this.replaceForms) || !this.replaceForms.length) {
+      console.warn(
+        `Configuration error. Config must define have a '${this.name}.${this.REPLACE_PROPERTY}' array defined.`
+      );
+
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * @param {Array<Doc>} docs - docs to be saved
+   * @return {Boolean} - whether any of the docs from the batch should be processed
+   */
+  filter(docs: Doc[]) {
+    return !!docs.filter(doc => doc.type === 'data_record').length;
+  }
+
+  /**
+   * @param {Array<Doc>} docs - docs to run the transition over
+   * @returns {Promise<Array<Doc>>} - updated docs (may include additional docs)
+   */
+  async run(docs: Doc[]) {
+    const originalContact = await this.createUserForContactsService.getUserContact();
+    if (!originalContact) {
+      return docs;
+    }
+
+    if (this.createUserForContactsService.isReplaced(originalContact)) {
+      this.reparentReports(docs, originalContact);
+    }
+    const userReplaceDoc = this.getUserReplaceDoc(docs);
+    if (userReplaceDoc) {
+      return this.replaceUser(docs, userReplaceDoc, originalContact);
+    }
+
+    return docs;
+  }
+
+  private getTransitionConfig(settings) {
+    return settings[this.name];
+  }
+
+  private getReplaceForms(settings) {
+    return this.getTransitionConfig(settings)?.[this.REPLACE_PROPERTY] || [];
+  }
+
+  private getUserReplaceDoc(docs: Doc[]) {
+    return docs.find(doc => this.replaceForms.includes(doc.form)) as UserReplaceDoc;
```

Comment:
Correct, it is possible for multiple forms to be configured and for more than one of them to be submitted in a given batch.

I will add an error there that gets thrown if that happens (since I cannot really think of any good reason we should add extra logic to try to support that behavior...)